SPECS/python-werkzeug/python-werkzeug.spec

```diff
@@ -19,36 +25,61 @@ BuildRequires:  python3-devel
 BuildRequires:  python3-libs
 BuildRequires:  python3-setuptools
 BuildRequires:  python3-xml
+BuildRequires:  python3-flit-core
 Requires:       python3
-%if 0%{?with_check}
+BuildRequires:  python3-pip
+%if %{with_check}
 BuildRequires:  curl-devel
 BuildRequires:  openssl-devel
-BuildRequires:  python3-pip
-BuildRequires:  python3-requests
 %endif
 
 %description -n python3-werkzeug
 Werkzeug started as simple collection of various utilities for WSGI applications and has become one of the most advanced WSGI utility modules. It includes a powerful debugger, full featured request and response objects, HTTP utilities to handle entity tags, cache control headers, HTTP dates, cookie handling, file uploads, a powerful URL routing system and a bunch of community contributed addon modules.
 
+%package -n python3-werkzeug-doc
+Summary:        Documentation for python3-werkzeug
+Requires:       python3-werkzeug = %{version}-%{release}
+ 
+%description -n python3-werkzeug-doc
+Documentation and examples for python3-werkzeug.
+
+%generate_buildrequires
+%if %{with_check}
+# -t picks test.txt by default which contains too tight pins
+%pyproject_buildrequires requirements/tests.in requirements/docs.in
+%else
+%pyproject_buildrequires -r requirements/docs.in
+%endif
+
 %prep
-%autosetup -n werkzeug-%{version} -p1
+%autosetup -p1 -n %{srcname}-%{version}
 
 %build
-%py3_build
+%pyproject_wheel
 
 %install
-%py3_install
+%pyproject_install
+%pyproject_save_files %{modname}
 
 %check
-pip3 install pytest hypothesis
-LANG=en_US.UTF-8 PYTHONPATH=./  python3 setup.py test
+pip3 install more-itertools exceptiongroup iniconfig tomli ephemeral_port_reserve pytest-xprocess MarkupSafe watchdog pytest requests cryptography
+%py3_check_import %{modname}
+%if %{with_check}
+# deselect the test_exclude_patterns test case as it's failing
+# when we set PYTHONPATH: https://github.com/pallets/werkzeug/issues/2404
+%pytest -Wdefault --deselect tests/test_serving.py::test_exclude_patterns \
+  --deselect tests/test_exceptions.py::test_response_body
```

Comment:
Choosing to skip this test. This test fails because the test somehow fails to call a package class constructor correctly: `missing ` required positional argument: `new_url`. 
![image](https://github.com/microsoft/azurelinux/assets/30801614/3794b47a-70c6-4d4b-b569-e0693eb8424d)

This class constructor is defined in: https://github.com/pallets/werkzeug/blob/817aac4d2c71b83752e451d23f96ac26cfd7bbcd/src/werkzeug/routing/exceptions.py#L36